File: akka-http-core/src/main/java/akka/http/javadsl/model/headers/Cookie.java
/**
 * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
 */

package akka.http.javadsl.model.headers;

/**
 *  Model for the `Cookie` header.
 *  Specification: https://tools.ietf.org/html/rfc6265#section-4.2
 */
public abstract class Cookie extends akka.http.scaladsl.model.HttpHeader {
    public abstract Iterable<HttpCookiePair> getCookies();

    public static Cookie create(HttpCookiePair... cookies) {
        return new akka.http.scaladsl.model.headers.Cookie(akka.http.impl.util.Util.<HttpCookiePair, akka.http.scaladsl.model.headers.HttpCookiePair>convertArray(cookies));
    }
    public static Cookie create(String name, String value) {
        return create(HttpCookiePair.create(name, value));
    }
}

File: akka-http/src/main/scala/akka/http/javadsl/server/values/Cookie.scala
/*
 * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
 */

package akka.http.javadsl.server.values

import akka.http.impl.server.{ RouteStructure, CookieImpl }
import akka.http.javadsl.server.{ Directive, RequestVal, Route }
import java.util.Optional

import scala.annotation.varargs
import scala.compat.java8.OptionConverters._

abstract class Cookie {
  def name(): String
  def domain(): Optional[String]
  def path(): Optional[String]

  def withDomain(domain: String): Cookie
  def withPath(path: String): Cookie

  def value(): RequestVal[String]
  def optionalValue(): RequestVal[Optional[String]]

  def set(value: String): Directive

  @varargs
  def delete(innerRoute: Route, moreInnerRoutes: Route*): Route =
    RouteStructure.DeleteCookie(name(), domain().asScala, path().asScala)(innerRoute, moreInnerRoutes.toList)
}
object Cookies {
  def create(name: String): Cookie = new CookieImpl(name)
}